### PR TITLE
Add integration tests for consistency of canAnnotate().

### PR DESCRIPTION
**Pull request by mtbc**
Adds a broken integration test to capture my surprise at https://trello.com/c/FAm5WmW8/24-how-to-query-object-permissions#comment-5c3da2a37d31124c5084f9cb.

### CODE REVIEW

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-broken/lastCompletedBuild/testngreports/integration/PermissionsTest/ failures should be at one of the new Assert lines.

File: components/tools/OmeroJava/test/integration/PermissionsTest.java [hunk 1]
```diff
@@ -114,4 +128,45 @@ public void testRootCanPermissionsByProjection() throws Exception {
         Assert.assertTrue(((RBool) projectPermsMap.get("canChgrp")).getValue());
         Assert.assertTrue(((RBool) projectPermsMap.get("canChown")).getValue());
     }
+
+    /**
+     * Test that an instance of the given class can be annotated only if the instance's {@link Permissions#canAnnotate()} is true.
+     * @param annotateeClass the class of which to try annotating an instance
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/PermissionsTest.java:134]: is it intended ``annotatee``?

Comment (mtbc) [reply components/tools/OmeroJava/test/integration/PermissionsTest.java:134]: "Annotatee" seemed the best succinct term for "annotated thing", i.e., the target of the annotation. I'm [not the first](http://www.iis.sinica.edu.tw/~swc/pub/information_leakage_socialnets.html) to coin it but admittedly it'll take a few more of us to get it into dictionaries!

**Review by joshmoore — APPROVED** [review]
As with the recent python test addition, intent seems clearly specified; happy to have this in. (I do wonder if some process in the future would help us detect which of these "broken" tests is still awaiting a fix)

**Comment by rgozim** [comment]
Code looks good. Seems like a good way to ensure annotate permission is synced with what the API behaviour.

**Pull request merged**